osbuild.spec: don't mangle shebangs

larskarlitski opened the pull request:
Stop the rpm post install script from mangling shebangs for runners,
stages, and assemblers. This breaks running them in osbuild containers,
which might contain a different operating systems than the host.

In practice, this means all build roots must have python3 explicitly
installed for most (all?) current stages and assemblers.

Fixes #354

@gicmo, @dvdhrm I couldn't figure out how to test this. Ideally, we'd have a couple of pipelines with nested `build` sections comprised of multiple combinations of operating systems.

@teg Do we always install `python3` from osbuild-composer? (I think we do, just making sure.)

larskarlitski commented:
The macro is defined (caution: Red Hat-internal link) in the [redhat-rpm-config package](http://pkgs.devel.redhat.com/cgit/rpms/redhat-rpm-config/tree/macros?h=rhel-8.4.0#n162).

teg commented:
We do not pull in python3. Shouldn't we rather make the rhel runners use platform-python explicitly?

gicmo commented:
> We do not pull in python3. Shouldn't we rather make the rhel runners use platform-python explicitly?

We are already doing that for the exact reason, see cb265e8b582ae3925411bd7bbeec9de61a986570

gicmo commented:
> @teg Do we always install `python3` from osbuild-composer? (I think we do, just making sure.)

I think this is done already: https://github.com/osbuild/osbuild-composer/blob/0f70edb48eaf399abad3a58132426e1d89dcb3eb/internal/distro/rhel8/distro.go#L1003

larskarlitski commented:
Force-pushed to clarify the explanation and link to the previous commit.

Thanks for the research @gicmo!

larskarlitski commented:
Force-pushed again to include a commit syncing `schutzbot`.

larskarlitski commented:
Force-pushed because `rpm` failed on having an `%install` in a comment. :angry:

larskarlitski commented:
`test_stages` started failing recently, because the kernel mirrors removed fedora-30. I've addressed this in #527. Let's wait for that to land before going forward with this one.

larskarlitski commented:
> `test_stages` started failing recently, because the kernel mirrors removed fedora-30. I've addressed this in #527. Let's wait for that to land before going forward with this one.

#527 landed. I've rebased this and it's ready for another look.

gicmo reviewed (approved):
:heart: perfect! Thanks so much!
